webpack.config.js

```diff
@@ -47,21 +53,86 @@ var config = {
       })
     ],
   module: {
-    rules: [
-      { test: /\.ts$/, use: "ts-loader" },
-      {
-        test: /\.ts$/,
-        enforce: "pre",
-        loader: "tslint-loader",
-        options: {
-          typeCheck: false,
-          configFile: false,
-          failOnHint: true
-        }
-      }
+    unknownContextRegExp: /$^/,
+    unknownContextCritical: false,
+    exprContextRegExp: /$^/,
+    exprContextCritical: false,
+    wrappedContextCritical: true,
+    rules: [{
+      test: /\.(jsx|js)$/,
+      use: [{
+        loader: "babel-loader"
+      }]
+    }, {
+      test: /\.(tsx|ts)$/,
+      loader: "ts-loader",
+      exclude: "/node_modules/"
+    }]
+  },
+  externals: process.env.NODE_ENV === "production" ? [nodeExternals()] : []
+};
+
+var configServer = {
+  entry: ["es6-shim", APP_DIR + "/index.ts"],
+  devtool: "cheap-module-source-map",
+  target: "node",
+  output: {
+    path: BUILD_DIR,
+    filename: "xcomponentapi.node.js",
```

Comment:
We can't use the same code in the api and detect the current environment ?
browser -> use window....
nodejs -> ws
In this case, we could use the same js distribution no ?